config/config_file.go

```diff
@@ -60,13 +73,14 @@ func New(filePath string) (*ConfigFile, error) {
 
 // EnsureFile creates the root dir and config file
 func EnsureFile() (string, error) {
+	setDefaults()
```

Comment:
@utsavanand2 this is still mutating a global constant.  From a practical perspective in the CLI, this is probably _ok_ here, but not a general pattern we would want to use.   Something like this simply references the global value when needed, but doesn't modify it

```go
permissions := DefaultPermissions
if IsRunningInCI() {
    permissions = DefaultCIPermissions
}
// ...
if err := os.MkdirAll(filepath.Dir(filePath), permissions); err != nil {
// ...
}
```

additionally, having a value for `defaultCIPermissions` means you can add a doc string explaining why we need that specific value in CI